src/components/ResourceList/ResourceList.tsx

```diff
@@ -611,39 +347,44 @@ class ResourceListInner extends React.Component<CombinedProps, State> {
       const overflow = viewportHeight - overlay.height;
 
       const spinnerHeight =
-        this.props.items.length === 1
-          ? SMALL_SPINNER_HEIGHT
-          : LARGE_SPINNER_HEIGHT;
+        items.length === 1 ? SMALL_SPINNER_HEIGHT : LARGE_SPINNER_HEIGHT;
 
       const spinnerPosition =
         overflow > 0
           ? (overlay.height - spinnerHeight) / 2
           : (viewportHeight - overlay.top - spinnerHeight) / 2;
 
-      this.setState({loadingPosition: spinnerPosition});
+      setLoadingPositionState(spinnerPosition);
     }
-  };
+  }, [listRef, items.length]);
 
-  private handleSelectAllItemsInStore = () => {
-    const {
-      onSelectionChange,
-      selectedItems,
-      items,
-      idForItem = defaultIdForItem,
-    } = this.props;
+  const itemsExist = useCallback(
+    (itemsParam?: ItemType[]) => {
+      return (itemsParam || items).length > 0;
+    },
+    [items],
+  );
 
-    const newlySelectedItems =
-      selectedItems === SELECT_ALL_ITEMS
-        ? getAllItemsOnPage(items, idForItem)
-        : SELECT_ALL_ITEMS;
+  useEffect(() => {
+    if (loading) {
+      setLoadingPosition();
+    }
+  }, [items.length, loading, setLoadingPosition]);
```

Comment:
I'm curious why we need this 2 hooks that do almost the same. I would think the following would be enough
```
 useEffect(() => {
    if(loading){
    setLoadingPosition();
   }
  }, [loading, setLoadingPosition]);
```

But it looks like the listRef is null download so bulkactions. It would be nice to rely on setLoadingPosition for the bulkactions to render. I wonder if we really need to check for listRef.current in the bulk actions check?

Reply:
It seems like I can get this to work if I do something like this:
```
!showEmptyState &&
    (showHeader || needsHeader) &&
    itemsExist() && (
      <div className={styles.HeaderOuterWrapper}>
        <Sticky boundingElement={listRef.current}>
```

instead of this

```
!showEmptyState &&
    (showHeader || needsHeader) &&
    listRef.current && (
```

Does that seem reasonable?

Reply:
I thought it did but I just tested it and basically we'll need the ref available before Sticky renders because it's essentially how the header knows when to stop sticking.  So when the list is scrolled off screen, the header stops sticking. Maybe this is a place for LazyRef that could trigger a state change?

Reply:
Everything about the code looks great! Just need to sort this last bit and I think @AndrewMusgrave was going to have a look today too.

Reply:
@athornburg I had a chat with @AndrewMusgrave and he might have a solution for this. So hold off on trying to fix for now.

Reply:
Sounds good. I was trying to figure this out last night, but I'm not sure I have enough context. Excited to hear what you all figure out.

Reply:
Hoping to take a look tonight! Will drop a comment tomorrow morning 😄